site/src/main/paradox/extras/AsyncDoFn.md

```diff
@@ -0,0 +1,41 @@
+# AsyncDoFn
+
+Scio's @scaladoc[`BaseAsyncDoFn`](com.spotify.scio.transforms.BaseAsyncDoFn) provides standard handling for sending asynchronous requests and capturing the responses for a bundle of pipeline elements.
+`BaseAsyncDoFn` is a subclass of @scaladoc[`DoFnWithResource`](com.spotify.scio.transforms.DoFnWithResource) which handles the creation and re-use of client classes.
+Scio provides several future-specific subclasses to choose from depending on the return type of the client:
+
+* @scaladoc[`GuavaAsyncDoFn`](com.spotify.scio.transforms.GuavaAsyncDoFn) for clients that return Guava's `ListenableFuture`
+* @scaladoc[`JavaAsyncDoFn`](com.spotify.scio.transforms.JavaAsyncDoFn) for clients that return `CompletableFuture`
+* @scaladoc[`ScalaAsyncDoFn`](com.spotify.scio.transforms.ScalaAsyncDoFn) for clients that return a scala `Future`
+
+`BaseAsyncDoFn` will wait for all futures for all bundle elements to be returned before completing the bundle.
+A failure of any request for an item in the bundle will cause the entire bundle to be retried.
+
+Given this mock client:
+```scala
+import com.google.common.util.concurrent.{ListenableFuture, Futures}
+
+case class MyClient(value: String) {
+  def request(i: Int): ListenableFuture[String] = Futures.immediateFuture(s"$value$i")
+}
+```
+
+A custom `DoFn` can be defined using `GuavaAsyncDoFn`, since the client returns `ListenableFuture`.
```

Comment:
```suggestion
We can invoke it in a Scio pipeline by implementing a `GuavaAsyncDoFn`, which supports the `ListenableFuture` interface.
```

Reply:
Trying to avoid "we" and "you" as a style preference, though sometimes unavoidable